tests/test_pagination.py

```diff
@@ -0,0 +1,132 @@
+
+    def test_get_url_params_previous_page_no_count(self):
+        pagination = self.create_pagination_with_all_properties_set()
+        pagination.count = None
+        url_params_previous_page_expected = {
+            client.Pagination.PARAM_OLDER_ID:
+                str(self._PAGINATION_OLDER_ID_CUSTOM),
+        }
+
+        self.assertTrue(pagination.has_previous_page())
+        self.assertEqual(url_params_previous_page_expected,
+                         pagination.url_params_previous_page)
+
+    def test_get_url_params_next_page_newer(self):
+        pagination = self.create_pagination_with_all_properties_set()
+        url_params_next_page_expected = {
+            client.Pagination.PARAM_COUNT: str(self._PAGINATION_COUNT_CUSTOM),
+            client.Pagination.PARAM_NEWER_ID:
+                str(self._PAGINATION_NEWER_ID_CUSTOM),
+        }
+
+        self.assertTrue(pagination.has_next_page_assured())
+        self.assertEqual(url_params_next_page_expected,
+                         pagination.url_params_next_page)
+
+    def test_get_url_params_next_page_newer_no_count(self):
+        pagination = self.create_pagination_with_all_properties_set()
+        pagination.count = None
+        url_params_next_page_expected = {
+            client.Pagination.PARAM_NEWER_ID:
+                str(self._PAGINATION_NEWER_ID_CUSTOM),
+        }
+
+        self.assertTrue(pagination.has_next_page_assured())
+        self.assertEqual(url_params_next_page_expected,
+                         pagination.url_params_next_page)
+
+    def test_get_url_params_next_page_future(self):
+        pagination = self.create_pagination_with_all_properties_set()
+        pagination.newer_id = None
+        url_params_next_page_expected = {
+            client.Pagination.PARAM_COUNT: str(self._PAGINATION_COUNT_CUSTOM),
+            client.Pagination.PARAM_NEWER_ID:
+                str(self._PAGINATION_FUTURE_ID_CUSTOM),
+        }
+
+        self.assertFalse(pagination.has_next_page_assured())
+        self.assertEqual(url_params_next_page_expected,
+                         pagination.url_params_next_page)
+
+    def test_get_url_params_next_page_future_no_count(self):
+        pagination = self.create_pagination_with_all_properties_set()
+        pagination.newer_id = None
+        pagination.count = None
+        url_params_next_page_expected = {
+            client.Pagination.PARAM_NEWER_ID:
+                str(self._PAGINATION_FUTURE_ID_CUSTOM),
+        }
+
+        self.assertFalse(pagination.has_next_page_assured())
+        self.assertEqual(url_params_next_page_expected,
+                         pagination.url_params_next_page)
+
+    def test_get_url_params_prev_page_from_pagination_with_no_prev_page(self):
+        pagination = self.create_pagination_with_all_properties_set()
+        pagination.older_id = None
+
+        def access_url_params_previous_page():
+            _ = pagination.url_params_previous_page
```

Comment:
Is there really a need to assign it to `_` ? Can't you just call `agination.url_params_previous_page` without assigning it ?

Reply:
Without assignment it logically gets an IDE warning. This is a Google-accepted way of suppressing these IDE warnings. See: https://google.github.io/styleguide/pyguide.html#Python_Language_Rules

Reply:
Fair enough